src/Skia/Avalonia.Skia/DrawingContextImpl.cs

```diff
@@ -620,26 +621,33 @@ public void Custom(ICustomDrawOperation custom)
         public void PushOpacityMask(IBrush mask, Rect bounds)
         {
             CheckLease();
-            // TODO: This should be disposed
-            var paint = new SKPaint();
+
+            var paint = SKPaintCache.Get();
```

Comment:
This SKPaint instance is now never disposed nor returned to the pool.

Reply:
Here is how I see the code flowing.

1. `PushOpacityMask` creates or gets an instance of `SKPaint` from the `SKPaintCache`. 
2. `CreatePaint` is used to create a `PaintWrapper` it to ensure all disposables are called later and then is pushed into the `_maskStack`.
3. `PopOpacityMask` is called and the `PaintWrapper` is poped off the `_maskStack`, disposed of when existing the using closure
4. The Wrapped `SKPaint` then is returned to the SKPaintCache for reuse.

![image](https://user-images.githubusercontent.com/224169/201951529-e7cc9d41-1755-4acd-aa3c-27f459961976.png)

I do not see any other uses of `_maskStack` outside of `PushOpacityMask` and `PopOpacityMask`.  If these two places are the only locations the `_maskStack` field is referenced, then one of two things will occur.
 - `PushOpacityMask` will be called (possibly multiple times) and then the corresponding number of `PopOpacityMask` will be called to pip the paints off the stack. - For this scenario, things work great and `SKPaint` objects are reused as much as possible.  This looks to be the normal flow,
 - `PushOpacityMask` will be called and then `PopOpacityMask` will not be called for some reason.  -  In this case, an extra SKPaint will be created and GCed with the `DrawingContextImpl` at some later time.